pkg/workflows/testutils/runner_test.go

```diff
@@ -16,82 +17,140 @@ import (
+		runner := testutils.NewRunner(tests.Context(t))
 
-		triggerMock := basictriggertest.Trigger(runner, func() (basictrigger.TriggerOutputs, error) {
-			return basictrigger.TriggerOutputs{CoolOutput: "cool"}, nil
-		})
-
-		actionMock := basicactiontest.Action(runner, func(input basicaction.ActionInputs) (basicaction.ActionOutputs, error) {
-			assert.True(t, input.InputThing)
-			return basicaction.ActionOutputs{AdaptedThing: "it was true"}, nil
-		})
-
-		consensusMock := ocr3test.IdenticalConsensus[basicaction.ActionOutputs](runner)
-
-		targetMock := chainwritertest.Target(runner, "chainwriter@1.0.0", func(input chainwriter.TargetInputs) error {
-			return nil
-		})
+		triggerMock, actionMock, consensusMock, targetMock := setupAllRunnerMocks(t, runner)
 
-		err := runner.Run(workflow)
-		require.NoError(t, err)
+		require.NoError(t, runner.Run(workflow))
 
 		trigger := triggerMock.GetStep()
-		assert.NotNil(t, trigger)
 		assert.NoError(t, trigger.Error)
 		assert.Equal(t, "cool", trigger.Output.CoolOutput)
 
 		action := actionMock.GetStep("action")
-		assert.NotNil(t, action)
+		assert.True(t, action.WasRun)
 		assert.Equal(t, basicaction.ActionInputs{InputThing: true}, action.Input)
 		assert.NoError(t, action.Error)
 		assert.Equal(t, "it was true", action.Output.AdaptedThing)
 
 		assert.True(t, helper.transformTriggerCalled)
-		consensus := consensusMock.GetStepDecoded("Consensus")
+		consensus := consensusMock.GetStepDecoded("consensus")
 		assert.Equal(t, "it was true", consensus.Output.AdaptedThing)
-		require.Len(t, consensus.Input, 1)
+		require.Len(t, consensus.Input.Observations, 1)
 
+		rawConsensus := consensusMock.GetStep("consensus")
 		target := targetMock.GetAllWrites()
-		require.NoError(t, err)
-		assert.Empty(t, target.Errors)
+		assert.Len(t, target.Errors, 0)
 		assert.Len(t, target.Inputs, 1)
-		assert.Equal(t, consensus.Output, target.Inputs[0].SignedReport)
+		assert.Equal(t, rawConsensus.Output, target.Inputs[0].SignedReport)
 	})
 
-	t.Run("Captures errors", func(t *testing.T) {
-		t.Skip("TODO https://smartcontract-it.atlassian.net/browse/KS-442")
-		assert.Fail(t, "Not implemented")
+	t.Run("Run allows hard-coded values", func(t *testing.T) {
+		workflow := workflows.NewWorkflowSpecFactory(workflows.NewWorkflowParams{Name: "tester", Owner: "ryan"})
+		trigger := basictrigger.TriggerConfig{Name: "trigger", Number: 100}.New(workflow)
+		hardCodedInput := basicaction.NewActionOutputsFromFields(workflows.ConstantDefinition("hard-coded"))
+		tTransform := workflows.Compute2[basictrigger.TriggerOutputs, basicaction.ActionOutputs, bool](
+			workflow,
+			"transform",
+			workflows.Compute2Inputs[basictrigger.TriggerOutputs, basicaction.ActionOutputs]{Arg0: trigger, Arg1: hardCodedInput},
+			func(SDK workflows.SDK, tr basictrigger.TriggerOutputs, hc basicaction.ActionOutputs) (bool, error) {
+				assert.Equal(t, "hard-coded", hc.AdaptedThing)
+				assert.NotNil(t, SDK)
+				assert.Equal(t, "cool", tr.CoolOutput)
```

Comment:
[This folder](https://github.com/smartcontractkit/chainlink-common/tree/main/pkg/capabilities/cli/cmd/testdata/fixtures/capabilities) has the ones that I refer to.  I used fun names because I didn't want to think to hard about real ones, since they are just for testing the engine and generator.

In particular, CoolOutput is defined [here](https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/capabilities/cli/cmd/testdata/fixtures/capabilities/basictrigger/basic_trigger-schema.json#L23), which generates the Go type, the capability's builder and the mock.